### PR TITLE
Fix skin tone sorting and refactor color procs

### PR DESCRIPTION
**Pull request by AndrewL97**
Add stack_trace to hex2rgb to catch future nonsense

## About The Pull Request

Remove lots of unused color code - replaced everything I could (be bothered to) with native byond calls
Fixed skin_tone getting lightness using a bad hex value - extra # added

## Why It's Good For The Game

Skin tones now sort properly, plus color code is less

image

## Changelog

:cl:
fix: skin tone sorting
refactor: refactor various color procs to use natives instead
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
hex2rgb existing makes me sad <https://secure.byond.com/docs/ref/#/proc/rgb2num>

**Comment by AndrewL97** [comment]
@LemonInTheDark Kyler just pointed me to this - I'll convert everything over

**Comment by AndrewL97** [comment]
@LemonInTheDark @Mothblocks 
I removed a bunch of unused procs - let me know if you'd like it reverted
but otherwise this should be good and dandy to review xoxoxo

**Review by Mothblocks — APPROVED** [review]

File: code/__HELPERS/type2type.dm [hunk 1]
```diff
@@ -152,71 +152,6 @@ GLOBAL_LIST_INIT(modulo_angle_to_dir, list(NORTH,NORTHEAST,EAST,SOUTHEAST,SOUTH,
...
 	return .
 
-//colour formats
-/proc/rgb2hsl(red, green, blue)
-	red /= 255;green /= 255;blue /= 255;
-	var/max = max(red,green,blue)
-	var/min = min(red,green,blue)
-	var/range = max-min
-
-	var/hue=0;var/saturation=0;var/lightness=0;
```
Comment (LemonInTheDark) [inline code/__HELPERS/type2type.dm:162]: impeccable

**Review by LemonInTheDark — APPROVED** [review]

File: code/__HELPERS/type2type.dm (hunk 1 above)
Comment (AndrewL97) [reply code/__HELPERS/type2type.dm:162]: image
I am really smart code man

**Review by Watermelon914 — APPROVED** [review]

**Pull request merged**